synapseclient/client.py

```diff
@@ -4575,14 +4576,26 @@ def submit(
         else:
             docker_repository = None
 
-        if "versionNumber" not in entity:
-            entity = self.get(entity, downloadFile=False)
-        # version defaults to 1 to hack around required version field and allow submission of files/folders
-        entity_version = entity.get("versionNumber", 1)
+        # Some hacks to make sure that new OOP models will also work with this interface
+        entity_is_dataclass = is_dataclass(entity)
```

Comment:
While this is fresh, I wonder if we should create a the tickets to add `Evaluation`, `Submission` to the OOP and how that will all work. 

1. We can add a spike ticket to organize our thoughts around the challenge OOP.
2. Investigate just how much of the client has left to be ported over to the OOP method

Reply:
https://sagebionetworks.jira.com/browse/SYNPY-1589
https://sagebionetworks.jira.com/browse/SYNPY-1590

I created these for Submission and Evaluation